src/combinatorics/simple_graph/adj_matrix.lean

```diff
@@ -15,89 +16,257 @@ properties to computational properties of the matrix.
 
 ## Main definitions
 
-* `adj_matrix` is the adjacency matrix of a `simple_graph` with coefficients in a given semiring.
+* `matrix.adj_matrix`: `A : matrix V V α` is qualified as an "adjacency matrix" if
+  (1) every entry of `A` is `0` or `1`,
+  (2) `A` is symmetric,
+  (3) every diagonal entry of `A` is `0`.
+
+* `matrix.adj_matrix.to_graph`: for `A : matrix V V α` and `h : adj_matrix A`,
+  `h.to_graph` is the simple graph induced by `A`.
+
+* `matrix.compl`: for `A : matrix V V α`, `A.compl` is supposed to be
+  the adjacency matrix of the complement graph of the graph induced by `A.adj_matrix`.
+
+* `simple_graph.adj_matrix`: the adjacency matrix of a `simple_graph`
+  with coefficients in a given semiring.
 
 -/
 
 open_locale big_operators matrix
 open finset matrix simple_graph
 
 universes u v
-variables {α : Type u}
-variables (R : Type v) [semiring R]
+variables {V α β R : Type u}
+
+namespace matrix
+
+/-- `A : matrix V V α` is qualified as an "adjacency matrix" if
+    (1) every entry of `A` is `0` or `1`,
+    (2) `A` is symmetric,
+    (3) every diagonal entry of `A` is `0`. -/
+structure adj_matrix [has_zero α] [has_one α] (A : matrix V V α) : Prop :=
+(zero_or_one : ∀ i j, (A i j) = 0 ∨ (A i j) = 1 . obviously)
+(symm : A.is_symm . obviously)
+(loopless : ∀ i, A i i = 0 . obviously)
+
+namespace adj_matrix
+
+lemma apply_diag [has_zero α] [has_one α]
+{A : matrix V V α} (h : adj_matrix A) (i : V) :
```

Comment:
```suggestion
  {A : matrix V V α} (h : adj_matrix A) (i : V) :
```
and make sure the indentation is like this for the other lemmas.

Reply:
Consider having `variables {A : matrix V V α}` at the beginning of this `namespace` section.